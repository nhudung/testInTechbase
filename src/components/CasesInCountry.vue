<template>
	<div>
		<template>
			<v-data-table
				:headers="headers"
				:items="data"
				:items-per-page="10"
				:sort-by.sync="sortBy"
				:sort-desc.sync="sortDesc"
			>
				<template v-slot:item.Date="{ item }">
					<span>{{ getDate(item.Date) }}</span>
				</template>
			</v-data-table>
		</template>
	</div>
</template>

<script>
	export default {
		props: {
			data: {
				type: Array,
				require: true,
			},
		},
		data() {
			return {
				headers: [
					{
						text: "Country",
						align: "center",
						value: "Country",
						sortable: true,
					},
					{ text: "Date", value: "Date", sortable: true },
					{ text: "City", value: "City" },
					{ text: "Confirmed", value: "Confirmed" },
					{ text: "Deaths", value: "Deaths" },
					{ text: "Recovered", value: "Recovered" },
					{ text: "Active", value: "Active" },
				],
				sortBy: "Date",
				sortDesc: true,
			};
		},
		methods: {
			getDate(date) {
				return date.slice(0, 10);
			},
		},
	};
</script>

<style></style>
