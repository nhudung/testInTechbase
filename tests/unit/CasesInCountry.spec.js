import { mount } from "@vue/test-utils";
import CasesInCountry from "@/components/CasesInCountry.vue";

describe("CasesInCountry.spec.js", () => {
	let cmp;

	beforeEach(() => {
		cmp = mount(CasesInCountry, {
			// Beaware that props is overriden using `propsData`
			propsData: {
				data: [
					{
						Country: "Viet Nam",
						CountryCode: "VN",
						Province: "",
						City: "",
						CityCode: "",
						Lat: "14.06",
						Lon: "108.28",
						Confirmed: 0,
						Deaths: 0,
						Recovered: 0,
						Active: 0,
						Date: "2020-01-22T00:00:00Z",
					},
				],
			},
		});
	});

	it('has received ["Cat"] as the message property', () => {
		expect(cmp.vm.data).toEqual([
			{
				Country: "Viet Nam",
				CountryCode: "VN",
				Province: "",
				City: "",
				CityCode: "",
				Lat: "14.06",
				Lon: "108.28",
				Confirmed: 0,
				Deaths: 0,
				Recovered: 0,
				Active: 0,
				Date: "2020-01-22T00:00:00Z",
			},
		]);
	});
	it("has the expected html structure", () => {
		expect(cmp.element).toMatchSnapshot();
	});
	it("is a CasesInCountry component", () => {
		expect(cmp.is(CasesInCountry)).toBe(true);
	});
	// Vue instance
	it("CasesInCountry is a vue instances", () => {
		expect(cmp.isVueInstance()).toBe(true);
	});
});
