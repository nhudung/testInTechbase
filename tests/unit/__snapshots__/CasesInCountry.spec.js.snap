// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CasesInCountry.spec.js has the expected html structure 1`] = `
<div>
  <v-data-table
    headers="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
    items="[object Object]"
    items-per-page="10"
    sort-by="Date"
    sort-desc="true"
  />
</div>
`;

exports[`Coutries.spec.js has the expected html structure 1`] = `
<div
  class="coutries-select"
>
  <v-overflow-btn
    editable=""
    item-text="Country"
    item-value="Slug"
    items="[object Object]"
    label="Viet Nam"
  />
</div>
`;
